tests/platform/test_reboot.py

```diff
@@ -21,42 +21,109 @@
 from check_interface_status import check_interface_status
 from check_transceiver_status import check_transceiver_basic
 from check_transceiver_status import all_transceivers_detected
+from psu_controller import psu_controller
 
+REBOOT_COMMAND = "command"
+REBOOT_CAUSE = "cause"
+REBOOT_TIMEOUT = "timeout"
+REBOOT_HELPER = "helper"
 
-def reboot_and_check(localhost, dut, interfaces, reboot_type="cold"):
+REBOOT_TYPE_WARMREBOOT = "warm"
+REBOOT_TYPE_COLDREBOOT = "cold"
+REBOOT_TYPE_FASTREBOOT = "fast"
+REBOOT_TYPE_REBOOT_BY_POWEROFF = "power off"
+REBOOT_TYPE_REBOOT_BY_WATCHDOG = "watchdog"
+
+reboot_ctrl_dict = {
+    REBOOT_TYPE_REBOOT_BY_POWEROFF : {
+        REBOOT_TIMEOUT : 300,
+        REBOOT_CAUSE : "Power Loss"
+    },
+    REBOOT_TYPE_COLDREBOOT : {
+        REBOOT_COMMAND : "reboot",
+        REBOOT_TIMEOUT : 300,
+        REBOOT_CAUSE : "reboot"
+    },
+    REBOOT_TYPE_FASTREBOOT : {
+        REBOOT_COMMAND : "fast-reboot",
+        REBOOT_TIMEOUT : 180,
+        REBOOT_CAUSE : "fast-reboot"
+    },
+    REBOOT_TYPE_WARMREBOOT : {
+        REBOOT_COMMAND : "warm-reboot",
+        REBOOT_TIMEOUT : 180,
+        REBOOT_CAUSE : "warm-reboot"
+    },
+    REBOOT_TYPE_REBOOT_BY_WATCHDOG : {
+        REBOOT_COMMAND : "python -c \"import sonic_platform.platform as P; P.Platform().get_chassis().get_watchdog().arm(5); exit()\"",
+        REBOOT_TIMEOUT : 300,
+        REBOOT_CAUSE : "Watchdog"
+    }
+}
+
+def check_reboot_cause(dut, reboot_cause_expected):
+    """
+    @summary: Check the reboot cause on DUT.
+    @param dut: The AnsibleHost object of DUT.
+    @param reboot_cause_expected: The expected reboot cause.
+    """
+    logging.info("Check the reboot cause")
+    output = dut.shell("show reboot-cause")
+    reboot_cause_got = output["stdout"]
+    logging.debug("show reboot-cause returns {}".format(reboot_cause_got))
+    m = re.search(reboot_cause_expected, reboot_cause_got)
+    assert m is not None, "got reboot-cause %s after rebooted by %s" % (reboot_cause_got, reboot_cause_expected)
+
+
+def reboot_and_check(localhost, dut, interfaces, reboot_type="cold", reboot_helper=None, reboot_argu=None):
```

Comment:
* Parameter passing should use pre-defined constant:
```
reboot_type=REBOOT_TYPE_COLD
```

* Suggest to use a better name `reboot_kwargs`.

Reply:
fixed